students/ZackConnaughton/lessons/lesson07/assignment/database.py

```diff
@@ -0,0 +1,234 @@
+
+                ]))
+        return(output_list)
+
+
+def print_mdb_collection(collection_name):
+    for doc in collection_name.find({}):
+        print(doc)
+
+def csv_list(file):
+    return_list = []
+    with open('sample_csv_files/' + file, encoding='utf-8-sig') as csv_file:
+        file_input = csv.DictReader(csv_file)
+        for row in file_input:
+            return_list.append(row)
+    return return_list
+
+def drop_database_data(collection):
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.media
+        database[collection].drop()
+        return True
+        #add something here if it cant be found
+
+
+class database_thread(Thread):
+
+    def __init__(self, q, collection_name, csv_file):
+        self.queue = q
+        self.collection_name = collection_name
+        self.csv_file = csv_file
+        super().__init__()
+
+    def run(self):
+        mongo = MongoDBConnection()
+        with mongo:
+            database = mongo.connection.media
+
+            database_collection = database[self.collection_name]
+            starting_count = database_collection.count()
+            database_list = csv_list(self.csv_file)
+            database_list_errors = 0
+            for item in database_list:
+                try:
+                    database_collection.insert_one(item)
+                except:
+                    database_list_errors += 1
+        output_list = (self.collection_name, database_collection.count() - starting_count, database_list_errors)
+        self.queue.put(output_list)
+        self.queue.task_done()
+        return None
+
+def format_output(output_raw):
+    data_entered = [0, 0, 0]
+    errors = [0, 0, 0]
+    for row in output_raw:
+        collection_name = row[0]
+        if collection_name == 'products':
+            data_entered[0] = row[1]
+            errors[0] = row[2]
+        if collection_name == 'customers':
+            data_entered[1] = row[1]
+            errors[0] = row[2]
+        if collection_name == 'rental':
+            data_entered[2] = row[1]
+            errors[0] = row[2]
+    print(data_entered)
+    print(errors)
+    return [data_entered, errors]
+
+
+def import_data(dir_name, product_csv, customer_csv, rental_csv):
+
+
+    data_entry_queue = Queue.Queue()
+
+    product_thread = database_thread(data_entry_queue, 'products', product_csv)
+    customer_thread = database_thread(data_entry_queue, 'customers', customer_csv)
+    rental_thread = database_thread(data_entry_queue, 'rental', rental_csv)
+
+    threads = []
+    threads.append(product_thread)
+    threads.append(customer_thread)
+    threads.append(rental_thread)
+
+    for thread in threads:
+        thread.start()
+
+    data_entry_queue.join()
+
+    output_raw = []
+    for item in range(len(threads)):
+        output_raw.append(data_entry_queue.get())
+
+    print(output_raw)
+    output = format_output(output_raw)
+    return output
```

Comment:
How well did your threads work? It looks pretty much how I would expect, but it depends on whether mongodb can thread if it saves any time. If it were network connections, there would probably be a more obvious gain. Nice job.